pagopa/io-app
# chore: [IOBP-971] Outcome screen for error 404 from server for pspList

## Short description
The key changes involve adding support for a new payment error type, `PAYMENT_METHOD_NOT_AVAILABLE_ERROR` to display an outcome when the api call for the list of psp is 404

## List of changes proposed in this pull request
- Updated `getPaymentAnalyticsEventFromRequestFailure` to handle `PAYMENT_METHOD_NOT_AVAILABLE_ERROR` 
- Added `PaymentMethodNotAvailableProblemJson` type and integrated it into the `WalletPaymentFailure` union type
- Updated `paymentsCalculatePaymentFeesAction` to include a cancel action and handle `notFound` errors
- Modified the reducer to handle the cancel action and map `notFound` errors to `PAYMENT_METHOD_NOT_AVAILABLE_ERROR`
- Send analytics to mixpanel 

## How to test
With the dev server mock a 404 response for the `calculateFees` function

- Start a payment 
- Select a payment method and continue
- Check if outcome result is correct. Tap on the button and check if it goes back to payment method selection

## Preview
https://github.com/user-attachments/assets/eabcdabc-8345-4b86-8602-21c21036e2a8
